integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/ExistsPredicateIT.java

```diff
@@ -30,6 +34,8 @@
 public class ExistsPredicateIT {
 
 	private static final String INDEX_NAME = "IndexName";
+	private static final String COMPATIBLE_INDEX_NAME = "CompatibleIndexName";
+	private static final String INCOMPATIBLE_INDEX_NAME = "IncompatibleIndexName";
```

Comment:
There are many ways for a field to have an incompatible definition in another index:

* object on one side, non-object on the other
* nested on one side, flattened on the other
* flattened with a given set of children on one side, object with a different set of children on the other
* flattened with same children on both side BUT the children themselves have incompatible exists predicates (e.g. doc values are enabled on one side but not the other).

I think that you should have at least one test for each of the cases above, even if you don't test all the possible combinations for the last case.